src/Analyser/TypeSpecifier.php

```diff
@@ -323,7 +323,8 @@ public function specifyTypesInCondition(
 					$argType = $scope->getType($expr->right->getArgs()[0]->value);
 					if ($argType->isString()->yes()) {
 						$accessory = new AccessoryNonEmptyStringType();
-						if ($leftType instanceof ConstantIntegerType && $leftType->getValue() >= 2) {
```

Comment:
the old code missed to handle `$offset` which turns into off-by-one errors on smaller/greater comparisons